File: app/src/main/java/phone/vishnu/todoapp/helper/NotificationHelper.java
package phone.vishnu.todoapp.helper;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;

import androidx.core.app.NotificationCompat;

import java.util.Calendar;

import phone.vishnu.todoapp.R;
import phone.vishnu.todoapp.activity.MainActivity;

import static androidx.core.app.NotificationCompat.DEFAULT_SOUND;
import static androidx.core.app.NotificationCompat.DEFAULT_VIBRATE;

public class NotificationHelper {

    private static final String NOTIFICATION_CHANNEL_ID = "phone.vishnu.shelvestodo";
    private static final String NOTIFICATION_CHANNEL_NAME = "TODONotificationChannel";
    private final Context context;
    private String title, description;

    public NotificationHelper(Context context, String title, String description) {
        this.context = context;
        this.title = title;
        this.description = description;
    }

    public void createNotification() {

        Calendar calendar = Calendar.getInstance();
        String timeString =
                String.valueOf(calendar.get(Calendar.MINUTE)) +
                        calendar.get(Calendar.HOUR_OF_DAY) +
                        calendar.get(Calendar.DAY_OF_MONTH) +
                        calendar.get(Calendar.MONTH);

        int NOTIFICATION_REQUEST_CODE = Integer.parseInt(timeString);

        Intent intent = new Intent(context, MainActivity.class);

        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

        final PendingIntent resultPendingIntent = PendingIntent.getActivity(context, NOTIFICATION_REQUEST_CODE, intent, 0);

        NotificationCompat.BigTextStyle bigTextStyle = new NotificationCompat.BigTextStyle();
        bigTextStyle.setBigContentTitle(title);
        bigTextStyle.bigText(description);


        final NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(context);
        mBuilder.setSmallIcon(R.drawable.ic_drawing)
                .setAutoCancel(true)
                .setContentIntent(resultPendingIntent)
                .setStyle(bigTextStyle)
                .setContentTitle(title)
                .setDefaults(DEFAULT_SOUND | DEFAULT_VIBRATE)
                .setPriority(Notification.PRIORITY_MAX);

        NotificationManager mNotificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            int importance = NotificationManager.IMPORTANCE_HIGH;

            NotificationChannel notificationChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID, NOTIFICATION_CHANNEL_NAME, importance);
            notificationChannel.setDescription("This is the notification channel for TODO Reminders");
            notificationChannel.setShowBadge(true);
            notificationChannel.setLockscreenVisibility(Notification.VISIBILITY_PUBLIC);

            mBuilder.setChannelId(NOTIFICATION_CHANNEL_ID);

            mNotificationManager.createNotificationChannel(notificationChannel);
        }
        assert mNotificationManager != null;
        mNotificationManager.notify(NOTIFICATION_REQUEST_CODE, mBuilder.build());

    }
}

File: app/src/main/java/phone/vishnu/todoapp/helper/NotificationHelper.kt
package phone.vishnu.todoapp.helper

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import androidx.core.app.NotificationCompat
import phone.vishnu.todoapp.MainActivity
import phone.vishnu.todoapp.R
import phone.vishnu.todoapp.model.Shelve

class NotificationHelper {

    companion object {
        fun createNotification(context: Context, shelve: Shelve?) {

            val requestCode = shelve?.dateDue!!.toInt()

            val intent =
                Intent(context, MainActivity::class.java).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

            val pendingIntent =
                PendingIntent.getActivity(
                    context,
                    requestCode,
                    intent,
                    PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
                )

            val builder =
                NotificationCompat.Builder(context, Constants.NOTIFICATION_CHANNEL_ID)
                    .setSmallIcon(R.drawable.ic_drawing)
                    .setContentTitle(shelve.title)
                    .setContentText(shelve.description)
                    .setContentIntent(pendingIntent)
                    .setAutoCancel(true)
                    .setDefaults(NotificationCompat.DEFAULT_SOUND or NotificationCompat.DEFAULT_VIBRATE)
                    .setPriority(NotificationCompat.PRIORITY_MAX)

            val notificationManager =
                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

            val notificationChannel = NotificationChannel(
                Constants.NOTIFICATION_CHANNEL_ID,
                Constants.NOTIFICATION_CHANNEL_NAME,
                NotificationManager.IMPORTANCE_HIGH
            )
            notificationChannel.description = "The notifications for TODO App Tasks"
            notificationChannel.setShowBadge(true)
            notificationChannel.lockscreenVisibility = Notification.VISIBILITY_PUBLIC
            builder.setChannelId(Constants.NOTIFICATION_CHANNEL_ID)
            notificationManager.createNotificationChannel(notificationChannel)

            notificationManager.notify(requestCode, builder.build())
        }
    }
}
